optuna_dashboard/llm/_backend.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from bottle import Bottle
+from bottle import request
+from bottle import response
+from .._bottle_util import json_api_view
+from ._prompt_templates import get_trial_filtering_prompt
+from .provider import InvalidAuthentication
+from .provider import RateLimitExceeded
+
+if TYPE_CHECKING:
+    from typing import TypedDict
+
+    from .llm.provider import LLMProvider
+
+    class TrialFilteringFuncResponse(TypedDict):
+        func_str: str
+        error_message: str
+
+
+def register_llm_route(app: Bottle, llm_provider: LLMProvider | None) -> None:
+    @app.post("/api/llm/trial_filter_query")
+    @json_api_view
+    def get_trial_filtering_func_str() -> dict[str, str]:
+        if llm_provider is None:
+            response.status = 400  # Bad Request
+            return {"reason": "No access to the LLM provider."}
+
+        user_query = request.json.get("user_query", "")
+        if not user_query:
+            response.status = 400  # Bad Request
+            return {"reason": "No user query provided."}
+
+        res: TrialFilteringFuncResponse | None = request.json.get("last_response", None)
+        if res is None:
+            prompt = get_trial_filtering_prompt(user_query)
+        else:
+            func_str = res.get("func_str", "")
+            err_msg = res.get("error_message", "")
+            prompt = get_trial_filtering_prompt(user_query, func_str, err_msg)
```

Comment:
nit: Can we simply write these lines like below?

```suggestion
        func_str = request.json.get("last_response", {}).get("func_str")
        error_msg = request.json.get("last_response", {}).get("error_message")
        prompt = get_trial_filtering_prompt(user_query, func_str, err_msg)
```

Reply:
Applied